src/middleware/node/get-payload.ts

```diff
@@ -11,20 +11,16 @@ import AggregateError from "aggregate-error";
 // }
 type IncomingMessage = any;
 
-export function getPayload(request: IncomingMessage): Promise<string> {
-  // If request.body already exists we can stop here
-  // See https://github.com/octokit/webhooks.js/pull/23
-
-  if (request.body) return Promise.resolve(request.body);
-
+export function getPayload(request: IncomingMessage): Promise<Buffer> {
   return new Promise((resolve, reject) => {
-    let data = "";
-
-    request.setEncoding("utf8");
+    let data: Buffer[] = [];
 
     // istanbul ignore next
     request.on("error", (error: Error) => reject(new AggregateError([error])));
-    request.on("data", (chunk: string) => (data += chunk));
-    request.on("end", () => resolve(data));
+    request.on("data", data.push.bind(data));
+    // istanbul ignore next
+    request.on("end", () =>
+      setImmediate(resolve, data.length === 1 ? data[0] : Buffer.concat(data)),
```

Comment:
why is the setImmediate needed here?

Reply:
I saw an explanation in this previous conversation, https://github.com/octokit/webhooks.js/pull/936#discussion_r1409945521

Reply:
@Uzlopak Can you please add a comment to the code about it?

Reply:
```suggestion
      // setImmediate improves the throughput by reducing the pressure from
      // the event loop
      setImmediate(resolve, data.length === 1 ? data[0] : Buffer.concat(data)),
```